src/main/java/net/snowflake/client/jdbc/ConnectionImplementationFactory.java

```diff
@@ -0,0 +1,20 @@
+package net.snowflake.client.jdbc;
+
+import java.sql.SQLException;
+import java.sql.SQLNonTransientConnectionException;
+import net.snowflake.client.core.SFSession;
+import net.snowflake.client.core.SFStatement;
+
+/**
+ * Factory class that presents the implementation of a Snowflake Connection. This allows for
+ * alternate definitions of SFSession, SFStatement, and SFResultSet, (representing the 'physical'
+ * implementation layer) that can share high-level code.
+ */
+public interface ConnectionImplementationFactory {
```

Comment:
It doesn't look like a factory pattern strictly speaking. It is more like an interface for `ConnectionImpl`.